src/core/args.c

```diff
@@ -769,13 +778,83 @@ void MVM_args_set_result_int(MVMThreadContext *tc, MVMint64 result, MVMint32 fra
     }
 }
 
+void MVM_args_set_result_uint(MVMThreadContext *tc, MVMuint64 result, MVMint32 frameless) {
+    MVMFrame *target;
+    if (frameless) {
+        target = tc->cur_frame;
+    }
+    else {
+        if (MVM_spesh_log_is_caller_logging(tc))
+            MVM_spesh_log_return_type(tc, NULL);
+        else if (MVM_spesh_log_is_logging(tc))
+            MVM_spesh_log_return_to_unlogged(tc);
+        target = tc->cur_frame->caller;
+    }
+    if (target) {
+        switch (target->return_type) {
+            case MVM_RETURN_VOID:
+                if (tc->cur_frame->static_info->body.has_exit_handler)
+                    save_for_exit_handler(tc,
+                        MVM_repr_box_int(tc, MVM_hll_current(tc)->int_box_type, result));
```

Comment:
Why isn't this `uint_box_type`?

Reply:
That's something I kinda hate about my uint changes, but also something that I don't really know a way around. uint_box_type is UInt which is not a proper type, but a subset. We need to know UInt so we can determine properties like most importantly signedness. But we cannot use it to box a value. I guess an alternative could be to split it into uint_box_type and uint_unbox_type.

Reply:
Oh right, ISTR some chat with lizmat about that a little while ago. Now I wonder if that's what was causing the problems I saw when I added an `hllboxtype_u` op and had some of the arg handling use that...